### ISSUE TITLE
Scoop-Search does not return powershell objects like the original command

### ISSUE THREAD

**Issue by Alfamari** [shilangyu/scoop-search#67]
## Description
Following the instructions of: `Behaves just like scoop search and returns identical output. If any differences are found please open an issue.`

Apart from being visually different, it also loses its ability for powershell object filtering/manipulation.

![wezterm-gui_401m3xovq8](https://github.com/user-attachments/assets/45fdf41a-c34d-4531-97e7-1dc0273a17b8)

![wezterm-gui_3WISRJT350](https://github.com/user-attachments/assets/ac6f8c17-b2e2-4732-ae45-823254a28459)

## Logs

The offending command causing the error with the `$env:SCOOP_SEARCH_VERBOSE=1` env var set:

<details>
  <summary>Output</summary>
  

```Commandline arguments: args.ParsedArgs{ .query = { 112, 121, 116, 104, 111, 110 }, .hook = false, .allocator = mem.Allocator{ .ptr = anyopaque@87975ff080, .vtable = mem.Allocator.VTable{ .alloc = fn (*anyopaque, usize, u8, usize) ?[*]u8@3e20d0, .resize = fn (*anyopaque, []u8, u8, usize, usize) bool@3e2170, .free = fn (*anyopaque, []u8, u8, usize) void@3e21c0 } } }
env:SCOOP=
env:USERPROFILE=C:\Users\alfamari
env:XDG_CONFIG_HOME=
Scoop config file path: C:\Users\alfamari/.config/scoop/config.json
Scoop config file contents: {
  "last_update": "2024-09-20T01:04:17.4222547-05:00",
  "scoop_repo": "https://github.com/ScoopInstaller/Scoop",
  "scoop_branch": "master",
  "root_path": "D:\\Program Files (x86)\\Scoop",
  "global_path": "D:\\Program Files (x86)\\Scoop",
  "aria2-enabled": true
}

Scoop home: D:\Program Files (x86)\Scoop
Found bucket: D:\Program Files (x86)\Scoop/buckets/extras
Worker count: 8
Found 5 matches
Found bucket: D:\Program Files (x86)\Scoop/buckets/main
Worker count: 8
Found 2 matches
Done searching
'extras' bucket:
    anaconda3 (2024.06-1) --> includes 'python.exe'
    fontforge (20230101) --> includes 'ffpython.exe'
    mambaforge (24.7.1-0) --> includes 'python.exe'
    miniconda3 (24.7.1-0) --> includes 'python.exe'
    paraview (5.13.0) --> includes 'pvpython.exe'

'main' bucket:
    python (3.12.6)
    winpython (3.12.4.1)
```
</details>

**Comment by Alfamari** [comment]
Additionally the output for returning no results is very slightly different. (Honestly, doesn't really seem noteworthy but I thought I'd document the difference and leave it up to the developer.)

![wezterm-gui_1AzxlzLxHY](https://github.com/user-attachments/assets/3e9c6b47-cdb6-4716-ae11-b311e250987e)

**Comment by Alfamari** [comment]
Well if it can output into a structured format, powershell does have ways of converting raw strings into objects:

Easiest method using standardized formats:
[ConvertFrom-Json (Microsoft.PowerShell.Utility) - PowerShell | Microsoft Learn](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/convertfrom-json?view=powershell-5.1)
[ConvertFrom-Csv (Microsoft.PowerShell.Utility) - PowerShell | Microsoft Learn](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/convertfrom-csv?view=powershell-5.1)

Can utilize customized non-standardized structured formats but much more annoying:
[ConvertFrom-String (Microsoft.PowerShell.Utility) - PowerShell | Microsoft Learn](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/convertfrom-string?view=powershell-5.1)
[ConvertFrom-StringData (Microsoft.PowerShell.Utility) - PowerShell | Microsoft Learn](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/convertfrom-stringdata?view=powershell-5.1)

**Comment by shilangyu** [comment]
I don't think that would be wanted though. If this were to be implemented I think it should directly output powershell objects

**Comment by shilangyu** [comment]
Commit in which scoop started returning PSObjects: https://github.com/ScoopInstaller/Scoop/commit/847756bd1e441979c121da8b20b67b56bc53814c

They also apply a change that returns all matches of binaries (#52). Interestingly, instead of using a PSObject for the list of binaries, they are concatenated using ` | ` into a string (why?).

**Comment by evaneliasyoung** [comment]
> I don't think that would be wanted though. If this were to be implemented I think it should directly output powershell objects

I've found a [StackOverflow question](https://stackoverflow.com/questions/30985008/create-powershell-psobject-in-c-sharp-cmdlet) which is on-topic, however implemented in C#.

[There is a way for C++ to target .NET](https://learn.microsoft.com/en-us/dotnet/core/porting/cpp-cli), however it:

- Cannot compile to an executable.
- is Windows only.

`PSObject` and `PSCustomObject` are [part of the .NET API](https://learn.microsoft.com/en-us/dotnet/api/system.management.automation.psobject?view=powershellsdk-7.4.0), and I'm not sure how/if the .NET API can be linked to Zig.

So much to say, I'm not sure this is possible in Zig, especially if we want to stay OS agnostic.

**Comment by goyalyashpal** [comment]
> If this were to be implemented I think it should directly output powershell objects

why??  

to me, outputting in a standard data format seems even more desirable than outputting in some particular language's internal object.


it was unix's SERIOUSLY SERIOUSLY missed opportunoty to have data streams.

**Comment by shilangyu** [comment]
> to me, outputting in a standard data format seems even more desirable than outputting in some particular language's internal object.

Of course, I totally agree. But since here we are slaves to the way scoop does it, powershell objects it is :/
